components/3001-call-undefined-component/error.txt

```diff
@@ -0,0 +1 @@
+Template "app-button" is used but it's not defined
```

Comment:
This should not be a compile-time error as the template may be defined in another file which might be compiled later or loaded dynamically.

It should not be a runtime error either, as it's possible that the server-side might want to render an empty HTML5 custom element which is then expanded client-side only.

Reply:
I was thinking of a runtime error or a new api call that would make that check with the currently compiled template.

I don't understand in what case you would choose to render an empty custom element.
What advantage would you get in this case ?

Reply:
I removed the test as you asked.
But i'm still interested in the use case suggested

Reply:
I currently have an undocumented feature in Magery that lets you call out to a JavaScript function for a given tag name. It will insert an empty tag into the DOM, then hand the element to your function, ignoring any child nodes. This has been useful in a real-world project where we have some jQuery widgets that do their own DOM manipulation.

I was going to link you to the docs and realised I haven't documented it yet! I'll get around to that. The API won't be final though as it still needs some lifecycle support (there's no destructor when the element is removed, for example).

Reply:
Obviously, on the server-side that custom JavaScript function will be unavailable, so just rendering an empty tag is fine (the JavaScript-only widget will pick it up later).

Reply:
Ahhh nice, I've seen that on when browsing js-magery code, didn't know what it was.
Cool